onnxruntime/test/framework/tensorutils_test.cc

```diff
@@ -109,31 +142,53 @@ static void CreateTensorWithExternalData(
 }
 
 template <typename T>
-static void TestUnpackExternalTensor(TensorProto_DataType type, const Path& model_path) {
-  // Create external data
-  std::basic_string<ORTCHAR_T> filename(ORT_TSTR("tensor_XXXXXX"));
-  TensorProto tensor_proto;
-  auto test_data = CreateValues<T>();
-  CreateTensorWithExternalData<T>(type, test_data, filename, tensor_proto);
-  std::unique_ptr<ORTCHAR_T, decltype(&DeleteFileFromDisk)> file_deleter(const_cast<ORTCHAR_T*>(filename.c_str()),
-                                                                         DeleteFileFromDisk);
-
+void UnpackAndValidate(const TensorProto& tensor_proto, const Path& model_path, const std::vector<T>& test_data) {
   // Unpack tensor with external data
   std::vector<T> val(test_data.size());
   auto st = utils::UnpackTensor(tensor_proto, model_path, val.data(), test_data.size());
   ASSERT_TRUE(st.IsOK()) << st.ErrorMessage();
 
   // Validate data
   for (size_t i = 0; i < test_data.size(); i++) {
-    ASSERT_EQ(val[i], test_data[i]);
+    ASSERT_TRUE(val[i] == test_data[i]);  // need to use ASSERT_TRUE with '==' to handle MFLoat16 and BFloat16
```

Comment:
why does ASSERT_EQ() not work?

Reply:
Error in gtest with it not knowing how to print the values if you use ASSERT_EQ. By using ASSERT_TRUE you avoid printing the value. We could add infrastructure so gtest could print MLFloat16/BFloat16 at some point.